src/prf_expand.rs

```diff
@@ -108,3 +109,18 @@ impl PrfExpand<([u8; 96], [u8; 32], [u8; 4])> {
     pub const SAPLING_ZIP32_CHILD_NON_HARDENED: Self = Self::new(0x12);
 }
 with_inputs!(a, A, b, B, c, C);
+
+impl PrfExpand<([u8; 32], [u8; 4], Option<(u8, VariableLengthSlice)>)> {
+    pub const ORCHARD_ZIP32_CHILD: Self = Self::new(0x81);
+    pub const ADHOC_ZIP32_CHILD: Self = Self::new(0xAB);
+    pub const REGISTERED_ZIP32_CHILD: Self = Self::new(0xAC);
+
+    /// Expands the given secret key in this domain.
+    pub fn with(self, sk: &[u8], a: &[u8; 32], b: &[u8; 4], c: Option<(u8, &[u8])>) -> [u8; 64] {
```

Comment:
nit: `with` is an odd name for this (as was the previous `with_tag`) in that this is actually applying the expansion operation.

Reply:
Also: is it intended that `Some((0, &[]))` be a valid input to this function?